ichnaea/rate_limit.py

```diff
@@ -0,0 +1,12 @@
+from redis import ConnectionError
+
+
+def rate_limit(redis_client, key, maxreq=0, expire=86400):
+    if maxreq:
+        try:
+            redis_client.set(key, 0, ex=expire, nx=True)
+            count = redis_client.incr(key, 1)
+            return count > maxreq
```

Comment:
I liked the former version with a pipeline better, that only made one network request instead of two. Calling `incr` on a non-existent key will just set it to 1, so the explicit "set to zero" isn't required.